src/roles.ts

```diff
@@ -17,3 +17,12 @@ export const sourceRoleRank: Record<SourceMemberRoles, number> = {
 };
 
 export const sourceRoleRankKeys = Object.keys(sourceRoleRank);
+
+export const rankToSourceRole = Object.entries(sourceRoleRank).reduce(
```

Comment:
Alternative something like this could maybe work?
But just a suggestion at this point.

```
const rankToSourceRole = (value) => {
  return Object.keys(sourceRoleRank).find(key => sourceRoleRank[key] === value);
}```

Reply:
Wanted to make a static map since it does not change during runtime and to mimic what we have for `sourceRoleRank`.